### PR TITLE
Fix issues with click effect not showing up on the side and top Nav when button is clicked

### PR DESCRIPTION
**Pull request by Sirmorrison**
When the Side and Top page navigation keys are pressed, due to the card overlay, the click button click effects are hidden.
This PR is to have this resolved.
This PR also adds a hover effects to the navigation buttons.
ezgif com-gif-maker(3)

### CODE REVIEW

**Review by dreacot — DISMISSED** [review]

File: ui/page/main_page.go [hunk 1]
```diff
@@ -46,11 +46,19 @@ type NavHandler struct {
...
-	isNavDrawerMinimized    bool
+	appBarNavItems []NavHandler
+	drawerNavItems []NavHandler
+
+	axis      layout.Axis
+	textSize  unit.Value
+	leftInset unit.Value
+	width     unit.Value
+	alignment layout.Alignment
+	direction layout.Direction
```
Comment (beansgum) [inline ui/page/main_page.go]: Create a NavDrawer struct in the components package to house the nav handlers and these properties.

File: ui/page/main_page.go [hunk 2]
```diff
@@ -601,19 +643,15 @@ func (mp *MainPage) LayoutNavDrawer(gtx layout.Context) layout.Dimensions {
...
-				}
-				return btn.Layout(gtx)
+				return mp.activeDrawerBtn.Layout(gtx)
 			})
 		}),
 	)
 }
 
-func (mp *MainPage) layoutCard(gtx layout.Context, background color.NRGBA, body layout.Widget) layout.Dimensions {
+func (mp *MainPage) layoutCard(gtx layout.Context, background color.NRGBA, Clickable *widget.Clickable, body layout.Widget) layout.Dimensions {
```
Comment (beansgum) [inline ui/page/main_page.go]: Tgus can be renamed to `layoutNavRow`.

File: ui/page/main_page.go [hunk 3]
```diff
@@ -537,59 +480,158 @@ func (mp *MainPage) LayoutUSDBalance(gtx layout.Context) layout.Dimensions {
...
+			)
+		}),
+		layout.Rigid(func(gtx layout.Context) layout.Dimensions {
+			gtx.Constraints.Min.X = gtx.Constraints.Max.X
+			return mp.Theme.Separator().Layout(gtx)
+		}),
+	)
+}
+
 func (mp *MainPage) LayoutNavDrawer(gtx layout.Context) layout.Dimensions {
```
Comment (beansgum) [inline ui/page/main_page.go]: The content of this function should also be part of the NavDrawer component.

**Review by beansgum — CHANGES_REQUESTED** [review]

File: ui/page/main_page.go [hunk 4]
```diff
@@ -316,6 +313,16 @@ func (mp *MainPage) Handle() {
 	}
 }
 
+func (mp *MainPage) defaultNavDims() {
```
Comment (beansgum) [inline ui/page/main_page.go]: This function should be in components/nav_drawer.go

File: ui/page/main_page.go [hunk 5]
```diff
@@ -258,20 +247,28 @@ func (mp *MainPage) UnlockWalletForSyncing(wal *dcrlibwallet.Wallet) {
...
 }
 
 func (mp *MainPage) Handle() {
-	for mp.minimizeNavDrawerButton.Button.Clicked() {
-		mp.isNavDrawerMinimized = true
+	mp.drawerNav.CurrentPage = mp.currentPageID()
+	mp.appBarNav.CurrentPage = mp.currentPageID()
+
+	for mp.drawerNav.MinimizeNavDrawerButton.Button.Clicked() {
+		mp.drawerNav.Axis = layout.Vertical
```
Comment (beansgum) [inline ui/page/main_page.go]: Add a toggle function in components/nav_drawer.go to switch between maximized and minimized attributes.

**Review by beansgum — CHANGES_REQUESTED** [review]

**Review by beansgum — APPROVED** [review]

**Review by dreacot — APPROVED** [review]

**Pull request merged**